src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/ApplicationPropertyPage.VisualBasic.xaml

```diff
@@ -36,7 +36,7 @@
        For VB projects, we expect to see every Form in the assembly directly or indirectly inherited from Form.
        We specify this by setting the SearchForEntryPointsInFormsOnly to true.
        We also want to ensure that the property always has a value. -->
-  <DynamicEnumProperty Name="StartupObject"
+  <DynamicEnumProperty Name="StartupObjectVB"
```

Comment:
You shouldn't need to do this, or change the visibility condition for either startup object property.

The "Application" `Rule` defined here extends the one in ApplicationPropertyPage.xaml; where the two files define properties of the same name, the one in this file replaces the other. Since this file should only be included for VB projects there's no need to check the language in the visibility condition.

If you're seeing multiple Application property pages, or multiple startup object properties in the UI then there's a bug we need to address, but not in the individual .xaml files.

Reply:
Oh, this is much better. Just tried it out and seems like it's all working fine!